models/classes/user/KvLtiUserService.php

```diff
@@ -66,11 +66,22 @@ protected function getPersistence()
      */
     protected function updateUser(LtiUser $user, LtiLaunchData $ltiContext)
     {
-        $user->setIdentifier(self::LTI_USER . $ltiContext->getUserID() . $ltiContext->getLtiConsumer()->getUri());
+        $technicalId = self::LTI_USER . $ltiContext->getUserID() . $ltiContext->getLtiConsumer()->getUri();
+
+        if (empty($user->getIdentifier())) {
+            $user->setIdentifier($technicalId);
+        }
+
+        $taoUserId = $user->getIdentifier();
+
         $this->getPersistence()->set(
             self::LTI_USER . $ltiContext->getUserID() . $ltiContext->getLtiConsumer()->getUri(),
```

Comment:
You can replace the entire `self::LTI_USER . $ltiContext->getUserID() . $ltiContext->getLtiConsumer()->getUri()` with ` $technicalId` I guess

Reply:
done